### DISCUSSION TITLE
Deprecation of formatting rules: `sort-keys` and `sort-vars`

### DISCUSSION THREAD

**Discussion by pubmikeb** [eslint/eslint#17692] [General] [answered]
According to the [Deprecation of formatting rules](https://eslint.org/blog/2023/10/deprecating-formatting-rules) post, starting the version 8.53, formatting rules will be deprecated.

In the post, we can find a long list of deprecated rules.
However, in this list, I could not find `sort-keys` and `sort-vars` rules, which, in my opinion, are also formatting rules, rather than code-construction rules.

Could you please clarify the criteria for labeling a rule as a formatting and non-formatting one?

**Comment by Rec0iL99** [comment, marked as answer]
Hi @pubmikeb, formatting rules are those rules that deal with whitespace. I believe it's rules that have [`type: layout`](https://github.com/eslint/eslint/blob/main/lib/rules/semi.js#L21C20-L21C20) in the rule metadata.

You can track #17522 for more insight on this decision.

**Reply by pubmikeb** [reply]
> formatting rules are those rules that deal with whitespace.

@Rec0iL99, however, [`comma-style`](https://eslint.org/docs/latest/rules/comma-style), [`quotes`](https://eslint.org/docs/latest/rules/quotes), and some others have nothing to do with whitespace.

**Reply by Rec0iL99** [reply]
Rules like `quotes` are handled by [prettier](https://prettier.io/docs/en/options#quotes) and [dprint](https://dprint.dev/plugins/typescript/config/#:~:text=rules.html%23semicolons-,quoteStyle,-How%20to%20use) and that's why they made it to the list. `comma-style` is indeed a whitespace related rule.
